email-common/src/main/java/io/commercetools/sunrise/email/EmailSender.java

```diff
@@ -44,15 +43,19 @@
      * }</pre>
      * Exceptions that occur while sending the email are instances of {@link EmailSenderException} that are
      * wrapped in an unchecked {@link CompletionException} thrown by {@link CompletableFuture#join()}.
-     * <p>Implementations of this method need to apply timeouts that may be configured when creating the {@link EmailSender}
+     * <p>
+     * Implementations of this method need to apply timeouts that may be configured when creating the {@link EmailSender}
      * instance. The timeouts avoid denial of service by too many connections waiting for stalled I/O.</p>
      *
-     * @param messageAuthor a consumer that fills the empty message created by the email sender.
-     *                      Note that {@link MimeMessage} instances are not immutable. Messages passed to the consumer
-     *                      must only be used by that consumer instance and must not be passed elsewhere.
+     * @param messageEditor the email sender passes an empty message to the message editor that the editor shall fill.
+     *                      Note that {@link MimeMessage} instances are not immutable. Messages passed to the editor
+     *                      must only be used by that editor instance and must not be passed elsewhere.
      * @return A completion stage for sending the message asynchronously.
+     * @throws EmailSenderException if there was an error while creating or filling the message. Note that in contrast
```

Comment:
The hierarchy `EmailCreationException extends EmailSenderException` and `EmailDeliveryException extends EmailSenderException` sounds good.

I'd choose `EmailDeliveryException` over `EmailTransmissionException` if there are objections.

> With the way @schleichardt showed you to wrap any exception inside the CompletionStage, let's go for it.

Just to make sure my understanding is correct: Do you mean that EmailSender.send(...) should neither throw EmailCreationException, nor EmailDeliveryException but instead wrap both kinds of exception in the returned CompletionStage?